client/src/game/temp.ts

```diff
@@ -90,7 +102,20 @@ export function addShape(shape: ApiShape, floor: string, layerName: LayerName, s
     if (sh === undefined) {
         return;
     }
+
+    // if this shape has been moved to a new location, notes are already set up for it client-side.
+    // we need to reattach the notes to this new shape since the local id is changed!
+    const noteList = Array.from(noteState.mutableReactive.notes.values());
```

Comment:
I don't think this needs mutable access, `noteState.reactive.notes.values())` should suffice.

Reply:
When trying to implement this I did originally use noteState.reactive but found that when passing `note` to `noteSystem.hookupShape(note, sh.id)` a TS2345 error was produced stating that readonly type is not assignable to a mutable type. So I was forced to use mutableReactive.

<details>
<summary>The full error</summary>

```
src/game/temp.ts:103:32 - error TS2345: Argument of type '{ readonly location: number | null; readonly text: string; readonly title: string; readonly creator: string; readonly uuid: string; readonly shapes: readonly GlobalId[]; readonly showOnHover: boolean; readonly showIconOnShape: boolean; readonly isRoomNote: boolean; readonly access: readonly { ...; }[]; readonly tags...' is not assignable to parameter of type 'ClientNote'.
  Type '{ readonly location: number | null; readonly text: string; readonly title: string; readonly creator: string; readonly uuid: string; readonly shapes: readonly GlobalId[]; readonly showOnHover: boolean; readonly showIconOnShape: boolean; readonly isRoomNote: boolean; readonly access: readonly { ...; }[]; readonly tags...' is not assignable to type 'Omit<ApiNote, "tags">'.
    Types of property 'shapes' are incompatible.
      The type 'readonly GlobalId[]' is 'readonly' and cannot be assigned to the mutable type 'GlobalId[]'.

103         noteSystem.hookupShape(note, sh.id);

```

</details>

Reply:
True, that would indeed error with the existing code; `hookupShape` however does not need a mutable reference to note for it to function, so its signature can be either modified to accept a `DeepReeadonly<ClientNote>` instead or what's probably better, only expect a note ID instead of a full note, as it only uses `note.uuid` in its code.

Could you update that?